pkg/standalone-utils/contracts/relayer/BaseRelayerLibrary.sol

```diff
@@ -118,6 +118,19 @@ contract BaseRelayerLibrary is IBaseRelayerLibrary {
      * @dev Returns true if `amount` is not actually an amount, but rather a chained reference.
      */
     function _isChainedReference(uint256 amount) internal pure override returns (bool) {
+        // First 3 nibbles are enough to determine if it's a chained reference.
+        return
+            (amount & 0xfff0000000000000000000000000000000000000000000000000000000000000) ==
+            0xba10000000000000000000000000000000000000000000000000000000000000;
+    }
+
+    /**
+     * @dev Returns true if `ref` is temporary reference, i.e. to be deleted after reading it.
+     */
+    function _isTemporaryChainedReference(uint256 amount) internal pure returns (bool) {
+        // First 3 nibbles determine if it's a chained reference.
+        // If the 4th nibble is 0 it is temporary; otherwise it is considered read-only.
+        // In practice, we shall use '0xba11' for read-only references.
```

Comment:
Note that the PR description says that e.g. `0xba2` would be read-only. I prefer the approach as shown in the code, and would update the description.